packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -65,10 +65,12 @@ export interface TCGCContext {
   __httpOperationExamples: Map<HttpOperation, SdkHttpOperationExample[]>;
   __pagedResultSet: Set<SdkType>;
   __mutatedGlobalNamespace?: Namespace; // the root of all tsp namespaces for this instance. Starting point for traversal, so we don't call mutation multiple times
+  __packageVersions?: string[]; // the package versions from the service versioning config and api version setting in tspconfig.
 
   getMutatedGlobalNamespace(): Namespace;
   getApiVersionsForType(type: Type): string[];
   setApiVersionsForType(type: Type, apiVersions: string[]): void;
+  getPackageVersions(): string[];
```

Comment:
Does it include those from `@clientApiVersions` and the filtering of `versioning -> previewStringRegex`?

PS: Should we have some docstring on the interface? This should be the most important interface of the TCGC, but there is zero doc.

Reply:
currently, `@clientApiVersions` and `previewStringRegex` only impact api version enum. i'm not sure if we should take it into account for all type's api version list. @iscai-msft for idea. also, idk if we should export `getPackageVersions` helper to emitter since it is only for inner usage now.

Reply:
Got it.

I don't have strong opinion on it. Just want to make sure we know that the values in it be (hence the comment on doc).

So, generally we just pick the last of the items, as the expected api-version.